src/main/java/Duke.java

```diff
@@ -1,10 +1,99 @@
-        System.out.println("Hello from\n" + logo);
+
+    public static String horizontalLine = "_".repeat(60) + "\n";
+    public static void main(String[] args) throws DukeException {
+        Scanner sc = new Scanner(System.in);
+        String task = "";
+        System.out.println("Hello! I'm Bot\n" + "What can I do for you?\n" + horizontalLine);
+        while (true) {
+            try {
+                task = sc.nextLine();
+                if (task.equals("bye")) {
+                    break;
+                }
+                if (task.equals("list")) {
+                    Task taskInstance = new Task("");
+                    taskInstance.printList();
+                    continue;
+                }
+                String[] elements = task.split((" "));
+                if (elements.length >= 2) {
+                    String instruction = elements[0];
+                    if (instruction.equals("mark") || instruction.equals("unmark")) {
+                        try {
+                            int no = Integer.parseInt(elements[1]);
+                            Task taskInstance = new Task("");
+                            if (instruction.equals("mark")) {
+                                taskInstance.mark(no);
+                            } else {
+                                taskInstance.unmark(no);
+                            }
+                            continue;
+                        } catch (NumberFormatException e) {
+                            System.err.println(Duke.horizontalLine + "You did not enter a valid integer :(\n" + Duke.horizontalLine);
+                            continue;
+                        }
+                    }
+                    if (instruction.equals("delete")) {
+                        try {
+                            int no = Integer.parseInt(elements[1]);
+                            Task taskInstance = new Task("");
+                            taskInstance.delete(no);
+                            continue;
+                        } catch (NumberFormatException e) {
+                            System.err.println(Duke.horizontalLine + "You did not enter a valid integer :(\n" + Duke.horizontalLine);
+                            continue;
+                        }
+                    }
+                    int firstSpaceIndex = task.indexOf(' ');
+                    String actualTask = task.substring(firstSpaceIndex + 1);
+                    if (instruction.equals("todo")) {
+                        Todo todo = new Todo(actualTask);
+                        todo.print();
+                    }
+                    else if (instruction.equals("deadline")) {
+                        String[] taskAndDeadline = actualTask.split(("/by"));
+                        if (taskAndDeadline.length == 1 || taskAndDeadline.length == 0) {
+                            throw new DukeException(Duke.horizontalLine + "OOPS!!! Invalid format for deadline :-(\n" + Duke.horizontalLine);
```

Comment:
Was wondering if this line is longer than 120 words (I notice this same issue in several other areas)?